arch/arm/core/cortex_m/prep_c.c

```diff
@@ -39,22 +39,40 @@ Z_GENERIC_SECTION(.vt_pointer_section) __attribute__((used)) void *_vector_table
 
 #ifdef CONFIG_CPU_CORTEX_M_HAS_VTOR
 
-#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
-
 /* In some Cortex-M3 implementations SCB_VTOR bit[29] is called the TBLBASE bit */
 #ifdef SCB_VTOR_TBLBASE_Msk
 #define VTOR_MASK (SCB_VTOR_TBLBASE_Msk | SCB_VTOR_TBLOFF_Msk)
 #else
 #define VTOR_MASK SCB_VTOR_TBLOFF_Msk
 #endif
 
+#ifdef CONFIG_SRAM_VECTOR_TABLE
+
+/* Vector table is in ROM, copy it to the RAM and set VTOR*/
+
+#define VECTOR_ADDRESS ((uintptr_t)_sram_vector_start)
+
+static inline void relocate_vector_table(void)
```

Comment:
Any particular reason you can't just leave this in .data and rely on the pre-existing copy code?